main.go

```diff
@@ -119,18 +127,47 @@ func main() { //nolint:funlen
 
 	ctrl.SetLogger(logger.ConfigLoggers(logmode))
 
+	cacheOptions := cache.Options{
+		Scheme: scheme,
+		ByObject: map[client.Object]cache.ByObject{
```

Comment:
Are all/most of the resources created by the operator labelled with some specific values ? if so, it could be possible to add additional selectors to fine tune watch mechanics,.

Reply:
currently controllers are watching different resources:
1. the ones created by Operator, these resources are mostly done by manifests, we use `CreateAddLabelsPlugin`  with defined labels.  But such labels are mainly to identify which component resource belongs to, not if this resource is part of ODH nor not. we will need to continue ADR to unify labeling first.
2. the ones not created by Operator, e.g namespaces, user could create new openshift project thro dashboard or even CLI